### PR TITLE
Go changelog

### PR DESCRIPTION
**Pull request by trodge**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by trodge** [comment]
Test comment

**Comment by trodge** [comment]
> Test comment

File: .github/workflows/check-changelog.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: check-changelog
```
Comment (ScottSuarez) [inline .github/workflows/check-changelog.yml:1]:
```suggestion
name: changelog-checker
```

File: .github/workflows/check-changelog.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+permissions: read-all
+
+on:
+  pull_request:
+    types: [opened, edited]
+
+jobs:
+  check-changelog:
```
Comment (ScottSuarez) [inline .github/workflows/check-changelog.yml:10]:
```suggestion
  check:
```

**Review by ScottSuarez — COMMENTED** [review]
Every find out whats up with the `.ci/changelog.tmpl`. Presumably this is still referencing that file?

**Comment by trodge** [comment]
> Every find out whats up with the `.ci/changelog.tmpl`. Presumably this is still referencing that file?

I believe that file is used in generating changelogs rather than checking them.

**Comment by ScottSuarez** [comment]
> > Every find out whats up with the `.ci/changelog.tmpl`. Presumably this is still referencing that file?
> 
> I believe that file is used in generating changelogs rather than checking them.

Any idea how it does the validation you were talking about. If `none` isn't a predefined type how is it recognizing it?

Could you try deleting the file `.ci/gcb-changelog-checker.yml`. I'm curious what will happen to the trigger in the advent of yml deletion. I am hoping a noop

**Comment by trodge** [comment]
> > > Every find out whats up with the `.ci/changelog.tmpl`. Presumably this is still referencing that file?
> > 
> > 
> > I believe that file is used in generating changelogs rather than checking them.
> 
> Any idea how it does the validation you were talking about. If `none` isn't a predefined type how is it recognizing it?
> 
> Could you try deleting the file `.ci/gcb-changelog-checker.yml`. I'm curious what will happen to the trigger in the advent of yml deletion. I am hoping a noop

I'm assuming that the container it uses is modified in some way.

When I removed the yaml file the step still ran and failed.

**Comment by ScottSuarez** [comment]
Can we change it to a noop and schedule a cleanup after some soak time?

ie
https://github.com/GoogleCloudPlatform/magic-modules/pull/8735/files

**Comment by ScottSuarez** [comment]
> > > > Every find out whats up with the `.ci/changelog.tmpl`. Presumably this is still referencing that file?
> > > 
> > > 
> > > I believe that file is used in generating changelogs rather than checking them.
> > 
> > 
> > Any idea how it does the validation you were talking about. If `none` isn't a predefined type how is it recognizing it?
> > Could you try deleting the file `.ci/gcb-changelog-checker.yml`. I'm curious what will happen to the trigger in the advent of yml deletion. I am hoping a noop
> 
> I'm assuming that the container it uses is modified in some way.
> 
> When I removed the yaml file the step still ran and failed.

So I take it you made those modifications in turn for this PR. I'll do a run through to make sure we captured everything. I'm sure you go it though

**Comment by trodge** [comment]
> Can we change it to a noop and schedule a cleanup after some soak time?
> 
> ie https://github.com/GoogleCloudPlatform/magic-modules/pull/8735/files

This seems to be causing a failure.

**Comment by ScottSuarez** [comment]
> > Can we change it to a noop and schedule a cleanup after some soak time?
> > ie [#8735 (files)](https://github.com/GoogleCloudPlatform/magic-modules/pull/8735/files)
> 
> This seems to be causing a failure.

Huh, it's not causing a failure on my test PR. Weird!!

Let's sync about how to clean this up.

**Comment by trodge** [comment]
> > > Can we change it to a noop and schedule a cleanup after some soak time?
> > > ie [#8735 (files)](https://github.com/GoogleCloudPlatform/magic-modules/pull/8735/files)
> > 
> > 
> > This seems to be causing a failure.
> 
> Huh, it's not causing a failure on my test PR. Weird!!
> 
> Let's sync about how to clean this up.

The current error is:
```
invalid build: invalid .steps field: no build steps are specified
```
This happens if I make `steps` unspecified or an empty array.

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**